spec/models/organization_spec.rb

```diff
@@ -91,6 +91,33 @@
     end
   end
 
+  context '#names' do
+    let(:organization) { Fabricate(:organization) }
+    let!(:raw_input_1) { Fabricate(:raw_input, import: Fabricate(:import, source: Fabricate(:source))) }
+    let!(:raw_input_2) { Fabricate(:raw_input, import: Fabricate(:import, source: Fabricate(:source))) }
+
+    it 'sorts names by created_at first then by rank' do
+      # ranks sequentially in the fabricator
+      rank1 = raw_input_1.source.rank_for(:organization_name_rank)
+      rank2 = raw_input_2.source.rank_for(:organization_name_rank)
+      expect(rank2).to eq rank1 + 1
+
+      PaperTrail.track_changes_with(raw_input_1) do
+        @n1 = Fabricate(:organization_name, organization: organization, created_at: Time.at(1))
+      end
+
+      PaperTrail.track_changes_with(raw_input_2) do
+        @n2 = Fabricate(:organization_name, organization: organization, created_at: Time.at(1))
+        @n3 = Fabricate(:organization_name, organization: organization, created_at: Time.at(2))
+      end
+
+      # n3 is created most recently
+      # n2 has a rank higher than n1, so it's last
+
+      expect(organization.names).to eq [@n3, @n1, @n2].map(&:content)
```

Comment:
as @jonallured describes, we would expect these sorted by `rank` first (ascending), `created_at` second. `n2` and `n3` come from a rank 2 source, with `n3` the newest, and `n1` comes from a rank 1 source. so the expected order is `[@n1, @n3, @n2]`, which is what I get when running this spec on master.

Reply:
Yeah, I think this might be a case of miscommunication about the requirements - we good to close this @dblock?

Reply:
I don't understand why this works, `organization_names.order(created_at: :desc).sort_by(&:rank)` isn't "sort by date in descending order, then re-sort by rank", it should be a sort by rank after some pre-sorted order by created_at, which can potentially re-shuffle everything, but somehow doesn't?

Reply:
Regardless of the requirements here [the result is not guaranteed to be stable](https://stackoverflow.com/questions/15442298/is-sort-in-ruby-stable). It just happened to be stable. Stable sort [was once requested](https://bugs.ruby-lang.org/issues/1089) to Ruby core a long time ago but was rejected for performance reasons.

Reply:
Ok, I see the issue now - not sure how to write a failing spec for it, but as it stands, it's possible for sort instability to produce incorrect tie breakers. I think you had it right here @dblock:

```ruby
organization_names.sort_by { |n| [n.rank, -n.created_at] }.pluck(:content)
```

That looks like what we want so that instability can't mess us up. Do you want to update this PR with that approach?

Just had a thought - what about this:

```ruby
organization_names.order(created_at: :desc).each_with_index.sort_by { |name, index| [name.rank, index] }.pluck(:content)
```

It's a lot more verbose, but helps me see that the original sort should be the tie breaker. 🤷‍♂️

Reply:
One more thought: regardless of how we fix this, it's probably worth a comment reminding Future Jon about sort instability so he doesn't refactor this back to the buggy version. 😝